kubelet/datadog_checks/kubelet/kubelet.py

```diff
@@ -487,6 +496,32 @@ def _submit_container_state_metric(self, metric_name, state_name, c_state, state
             gauge_name = '{}.containers.{}.{}'.format(self.NAMESPACE, metric_name, state_name)
             self.gauge(gauge_name, 1, tags + reason_tags)
 
+    def _report_ephemeral_storage_usage(self, pod_list, instance_tags):
+        stats = self._retrieve_stats()
+
+        ephemeral_storage_usage = {}
+        for pod in stats.get('pods', []):
+            pod_uid = pod.get('podRef', {}).get('uid')
+            pod_ephemeral_usage = pod.get('ephemeral-storage', {}).get('usedBytes')
+            if pod_uid and pod_ephemeral_usage:
+                ephemeral_storage_usage[pod_uid] = pod_ephemeral_usage
+
+        for pod in pod_list['items']:
```

Comment:
I know that we're assuming that we always have in `items` entry in the pod list, but I'm pretty sure that we had some issues (in the Go code) related to getting empty/nil pod list from the kubelet in some cases. @mfpierre could confirm that I think.
Let's handle it the same way we handle `pods` in the `stats` map.

Reply:
nevermind, we handle that here https://github.com/DataDog/integrations-core/blob/master/kubelet/datadog_checks/kubelet/kubelet.py#L269